src/App.js

```diff
@@ -11,88 +11,62 @@ import CoinsPage from "./pages/CoinsPage";
 import { PortfolioPage } from "./pages/PortfolioPage";
 import { GlobalStyle, StyledContainer } from "./ui";
 import { Navbar } from "./components/Navbar";
+import { changeCurrency } from "./store/currency/currency.actions";
 
-class App extends React.Component {
-  state = {
-    currencyDetails: {
-      currency: "",
-      currencySymbol: "",
-    },
-  };
-
-  componentDidMount() {
-    const savedData = JSON.parse(localStorage.getItem("currencyDetails"));
-    if (savedData !== null) {
-      this.setState({ currencyDetails: savedData });
-    } else {
-      this.setState({
-        currencyDetails: { currency: "usd", currencySymbol: "$" },
-      });
-    }
-  }
 
-  handleCurrencyChange = ({ target: { value } }) => {
-    const symbols = {
-      usd: "$",
-      gbp: "£",
-      eur: "€",
-      btc: "₿",
-      eth: "Ξ",
-    };
+function App(props) {
+  const {changeCurrency} = props;
+  const { currency, currencySymbol } = useSelector(
+    (state) => state.currencyDetails
+  );
 
-    this.setState({
-      currencyDetails: {
-        currency: value,
-        currencySymbol: symbols[value.toLowerCase()],
-      },
-    });
+  const handleCurrencyChange = ({ target: { value } }) => {
+    changeCurrency(value);
   };
 
-  render() {
-    const {
-      currencyDetails: { currency, currencySymbol },
-    } = this.state;
-    return (
-      <>
-        {currency && currencySymbol && (
-          <Router>
-            <GlobalStyle />
-            <StyledContainer>
-              <Navbar
-                currency={currency}
-                currencySymbol={currencySymbol}
-                handleCurrencyChange={this.handleCurrencyChange}
-              />
+  return (
+    <>
+      {currency && currencySymbol && (
```

Comment:
are you sure this is right? I don't like this... 

Navbar can have a selector inside, there is no need to have app state and pass it down